v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/DbConfigContainer.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates;
+
+import com.google.cloud.teleport.v2.source.reader.io.IoWrapper;
+import com.google.cloud.teleport.v2.spanner.migrations.schema.ISchemaMapper;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.transforms.Wait;
+
+/** Container for DB Configuration. */
+public interface DbConfigContainer {
```

Comment:
@bharadwaj-aditya 
Thanks.
As you had suggested in the review sync, I have kept the interface free of inheritance.
(This definitely improves the code cleanliness here)

Reply:
Can you elaborate on the java docs here.

The purpose of this class is to hold the relevant config to invoke migration of a given shard / Table. May make sense to clarify.

Reply:
Added a note.